### DISCUSSION TITLE
New Redis Licensing Changes

### DISCUSSION THREAD

**Discussion by Fryguy** [sidekiq/sidekiq#6237] [General]
I hadn't seen this shared with the Sidekiq community, so here it is: https://redis.com/blog/redis-adopts-dual-source-available-licensing/

> Beginning today, all future versions of Redis will be released with source-available licenses. Starting with Redis 7.4, Redis will be dual-licensed under the [Redis Source Available License (RSALv2)](https://redis.com/legal/rsalv2-agreement/) and [Server Side Public License (SSPLv1)](https://redis.com/legal/server-side-public-license-sspl/). Consequently, Redis will no longer be distributed under the three-clause Berkeley Software Distribution (BSD).

I don't have any personal thoughts on it really, but I do know the recent changes in many open source projects to BSL or similar licenses has been affecting the open source ecosystem.

See also https://github.com/redis/redis/pull/13157

**Comment by sobrinho** [comment]
@mperham I got a call with Redis Cloud team today and they scared me a little bit by saying that in the future no other cloud providers other than them will be able to offer managed Redis.

I'm not 100% sure that it's entirely accurate but given it was a statement from the Redis Cloud team it makes me wonder what's happening there.

Any ideas here?

**Reply by mperham** [reply]
That's an outright lie. Never believe the sales folks.

**Reply by mperham** [reply]
Going forward, no one can offer managed Redis versions **past 7.2** without a contract with Redis.com. But Sidekiq will work with Redis 7.2 for the foreseeable future because of this.

**Reply by mperham** [reply]
And the PaaS folks are all rapidly moving to Valkey and other forks as a managed Redis-compatible service. I expect they will all gather around Redis 7.2 as a standard baseline set of functionality that Sidekiq and you can depend on.

**Reply by sobrinho** [reply]
Good to know!

I had a call with Dragonfly a few months ago and they look good but I didn't do more than a "hello world" here and there to test. I stopped after figuring they don't actually support macOS. It runs on Docker so it's an additional friction but not a complete deal breaker.

I wonder if Redis can restrict the Redis protocol usage itself 🤔
